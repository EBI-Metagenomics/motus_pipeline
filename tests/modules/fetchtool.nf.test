nextflow_process {

    name "Test Process FETCHTOOL"
    script "modules/fetchtool.nf"
    process "FETCHTOOL"

    test("Should download single end reads") {

        when {
            params {
                // define parameters here. Example:
                outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = "SRR2187576"
                input[1] = ""
                input[2] = ""
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }


    test("Should download paired end reads") {

        when {
            params {
                // define parameters here. Example:
                outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = "ERR4387386"
                input[1] = ""
                input[2] = ""
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
