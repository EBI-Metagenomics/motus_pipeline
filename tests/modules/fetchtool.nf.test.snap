{
    "Should download single end reads": {
        "content": [
            {
                "reads": [
                    "SRR2187576.fastq.gz:md5,421386641e05978ba61550b851855743"
                ]
            }
        ],
        "timestamp": "2023-07-18T22:54:31+0000"
    },
    "Should download paired end reads": {
        "content": [
            {
                "reads": [
                    [
                        "ERR4387386_1.fastq.gz:md5,eb626d41e20ec61318d4541dbe84fdbe",
                        "ERR4387386_2.fastq.gz:md5,8cd92f17190c2c60b2651536f7eb11c6"
                    ]
                ]
            }
        ],
        "timestamp": "2023-07-18T22:54:31+0000"
    }
}
